app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/Message.java

```diff
@@ -26,12 +26,13 @@
     private String mType;
 
     @SerializedName("content")
-    private MessageContent mContent;
+    private T mContent;
 
     public static final String TEXT = "Text";
     public static final String LAT_LONG = "LatLong";
+    public static final String USER_LOCATION = "UserLocation";
```

Comment:
Lets use StringDef annotation to enumerate the message types and enforce it at "source" level (compile-time only).